pandas/core/reshape/reshape.py

```diff
@@ -1094,6 +1097,207 @@ def get_empty_frame(data) -> DataFrame:
+    dropped_first: None | Hashable | dict[str, Hashable] = None,
+) -> DataFrame:
+    """
+    Create a categorical `DataFrame` from a `DataFrame` of dummy variables.
+
+    Inverts the operation performed by 'get_dummies'.
+
+    Parameters
+    ----------
+    data : `DataFrame`
+        Data which contains dummy-coded variables.
+    subset : None, Index, or list of Hashables, default 'None'
+        The columns which to convert from dummy-encoding and return as categorical
+        `DataFrame`. If `columns` is None then all dummy columns are converted and
+        appended to the non-dummy columns.
+    sep : str or dict of str, default '_'
+        Separator used in the column names of the dummy categories they are
+        character indicating the separation of the categorical names from the prefixes.
+        For example, if your column names are 'prefix_A' and 'prefix_B',
+        you can strip the underscore by specifying sep='_'.
+        Alternatively, pass a dictionary to map prefix separators to prefixes if
+        multiple and/or mixed separators are used in the column names.
+    dropped_fist : None, Hashable or dict of Hashables, default None
+        The implied value the dummy takes when all values are zero.
+        Can be a a single value for all variables or a dict directly mapping the
+        dropped value to a prefix of a variable.
+
+    Returns
+    -------
+    `DataFrame`
+        Categorical data decoded from the dummy input-data.
+
+    See Also
+    --------
+    get_dummies : Convert `Series` or `DataFrame` to dummy codes.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame({"a": [1, 0, 0, 1], "b": [0, 1, 0, 0],
+    ...                   "c": [0, 0, 1, 0]})
+
+    >>> pd.from_dummies(df)
+    0     b
+    1     c
+    2     a
+
+    >>> df = pd.DataFrame({"col1_a": [1, 0, 1], "col1_b": [0, 1, 0],
+    ...                   "col2_a": [0, 1, 0], "col2_b": [1, 0, 0],
+    ...                   "col2_c": [0, 0, 1]})
+
+    >>> pd.from_dummies(df, sep="_")
+        col1    col2
+    0    a       b
+    1    b       a
+    2    a       c
+
+    >>> df = pd.DataFrame({"col1_a": [1, 0, 0], "col1_b": [0, 1, 0],
+    ...                    "col2_a": [0, 1, 0], "col2_b": [1, 0, 0],
+    ...                    "col2_c": [0, 0, 0]})
+
+    >>> pd.from_dummies(df, sep="_", dropped_first={"col1": "d", "col2": "e"])
+        col1    col2
+    0    a       b
+    1    b       a
+    2    d       e
+
+    >>> df = pd.DataFrame({"col1_a-a": [1, 0, 1], "col1_b-b": [0, 1, 0],
+    ...                   "col2-a_a": [0, 1, 0], "col2-b_b": [1, 0, 0],
+    ...                   "col2-c_c": [0, 0, 1]})
+
+    >>> pd.from_dummies(df, sep={"col1": "_", "col2": "-"})
+       col1  col2
+    0  a-a   b-b
+    1  b-b   a-a
+    2  a-a   c-c
+    """
+    from pandas.core.reshape.concat import concat
+
+    if subset is None:
+        subset = list(data.columns)
+    elif isinstance(subset, Index):
+        subset = list(subset)
```

Comment:
Using `list()` for now as `.tolist()` does not have typed returns, I will open a PR and fix these lines once the annotations are merged.

Reply:
list of an Index is well typed, so not sure what you mean